core/src/test/java/org/apache/iceberg/rest/auth/TestOAuth2Manager.java

```diff
@@ -63,10 +63,6 @@ void initSessionNoOAuth2Properties() {
     try (OAuth2Manager manager = new OAuth2Manager("test");
         OAuth2Util.AuthSession session = manager.initSession(client, properties)) {
       assertThat(session.headers()).isEmpty();
-      assertThat(manager)
```

Comment:
see my other comment (https://github.com/apache/iceberg/pull/12563#discussion_r2177618295), so we should be rolling back the changes here